java/ql/src/experimental/semmle/code/java/PathSanitizer.qll

```diff
@@ -20,13 +20,25 @@ private class ExactStringPathMatchGuard extends PathTraversalBarrierGuard instan
   }
 }
 
+/**
+ * Returns the qualifier of a method call if it's a variable access, or the qualifier of the qualifier if
+ * the qualifier itself is a method call to `getPath`, which helps to reduce FPs by handling scenarios
+ * such as `!uri.getPath().contains("..")`.
+ */
+private Expr getRealQualifier(Expr e) {
+  e.(MethodAccess).getMethod().hasQualifiedName("android.net", "Uri", "getPath") and
+  result = e.(MethodAccess).getQualifier()
+  or
+  result = e.(VarAccess)
```

Comment:
I would drop the `getPath` check, I think it's too specific por this library.

IMO, in this case we'll always want a `VarAccess` so that the guard actually does something — so traversing the qualifiers while they are `MethodAccess`es until we reach a `VarAccess` should make sense.

```suggestion
  result = getRealQualifier(e.(MethodAccess).getQualifier())
  or
  result = e.(VarAccess)
```

The QLDoc should be changed accordingly.

Reply:
Thanks for the advice. I've made the change.